app/lib/katello/resources/candlepin/pool.rb

```diff
@@ -18,8 +18,12 @@ def create(owner_key, attrs)
           end
 
           def find(pool_id)
-            pool_json = self.get(path(pool_id), self.default_headers).body
-            fail ArgumentError, "pool id cannot contain ?" if pool_id["?"]
+            begin
+              pool_json = self.get(path(pool_id), self.default_headers).body
+            rescue RestClient::ResourceNotFound
```

Comment:
When I pick this downstream I'll bring in the new CandlepinNotRunning error introduced in https://github.com/Katello/katello/pull/9270 to make sure this error is raised accurately

Reply:
Why not do it here?  Are upstream and downstream diverged?

Reply:
The referenced PR was merged 6 days ago so it hasn't reached any downstream build

Reply:
Oh, I understand your question now! My point was not that we would rescue from `CandlepinNotRunning` here. This code already behaves correctly because that error + supporting code is in master. The result is that if candlepin is down, CandlepinNotRunning is raised, whereas before it would raise  `ResourceNotFound` which is ambigious. Is candlepin down or is the pool gone? Having the `CandlepinNotRunning` error + supporting code is necessary to include downstream because in the event that candlepin _were_ down we could end up deleting all the pools with the new event. We can trust ResourceNotFound to be raised and caught here only when the pool is actually gone

Reply:
Makes sense!  Maybe there should be a test that fails when `CandlepinNotRunning` is not properly in place, then?

Reply:
Maybe but I don't think this PR is the place for it